tests/test_config_registry_and_test_execution.py

```diff
@@ -310,184 +314,91 @@ def test_config_registry_and_test_execution_failure_and_error_handling():
-        ),
-        (
-            {},  # tc_tags
-            {},  # cli_tags
-            True,  # is_test_running
-        ),
-        (
-            {
-                "variant": [
-                    "omicron",
-                ],
-                "branch-level": [
-                    "some_branch",
-                ],
-            },  # tc_tags
-            {
-                "branch_level": "some_branch",
-            },  # cli_tags
-            True,  # is_test_running
-        ),
-        (
-            {
-                "variant": [
-                    "omicron",
-                ],
-                "branch_level": [],
-            },  # tc_tags
-            {
-                "variant": "omicron",
-            },  # cli_tags
-            True,  # is_test_running
-        ),
-        (
-            {
-                "variant": [
-                    "omicron",
-                ],
-                "branch_level": [
-                    "leaf",
-                ],
+                "k1": ["v1", "v3"],
+                "k2": ["v2", "v5", "v6"],
             },  # tc_tags
-            {
-                "branch_level": "leaf",
-            },  # cli_tags
-            True,  # is_test_running
+            {"k1": "v2"},  # cli_tags
+            True,  # expect_skip
+            id="one tag name matches but no value",
         ),
-        (
-            {
-                "variant": [
-                    "omicron",
-                ],
-                "branch_level": [
-                    "leaf",
-                ],
-            },  # tc_tags
-            {
-                "branch_level": "bud",
-            },  # cli_tags
-            False,  # is_test_running
-        ),
-        (
+        pytest.param(
             {
                 "k1": ["v1", "v3"],
                 "k2": ["v2", "v5", "v6"],
             },  # tc_tags
-            {
-                "k1": "v1",
-                "k2": "v2",
-            },  # cli_tags
-            True,  # is_test_running
+            {"k1": "v2", "k2": ["v12", "v13"]},  # cli_tags
+            True,  # expect_skip
+            id="multple tag names match but no value",
         ),
-        (
+        pytest.param(
             {
-                "k1": ["v1", "v7"],
+                "k1": ["v1", "v3"],
                 "k2": ["v2", "v5", "v6"],
             },  # tc_tags
             {
                 "k1": "v1",
                 "k2": "v2",
             },  # cli_tags
-            True,  # is_test_running
+            False,  # expect_skip
+            id="multiple tag names match and multiple values",
         ),
-        (
+        pytest.param(
             {
                 "k1": ["v1", "v7"],
                 "k2": ["v2", "v5", "v6"],
             },  # tc_tags
-            {
-                "k1": "v1",
-                "k2": "v10",
-            },  # cli_tags
-            True,  # is_test_running
+            {"k3": "v10"},  # cli_tags
+            False,  # expect_skip
```

Comment:
I would have expect skip true when a testcase doest have the selected key.

Reply:
This is now the case